libewf/libewf_single_file_extended_attribute.h

```diff
@@ -0,0 +1,153 @@
+	 */
+	uint8_t type;
+
+	/* The name
+	 */
+	uint8_t *name;
+
+	/* The name size
+	 */
+	size_t name_size;
+	
+	/* The value
+	 */
+	uint8_t *value;
+
+	/* The value size
+	 */
+	size_t value_size;
+
+	/* The item level
+	 */
+	int level;
+};
+
+int libewf_single_file_extended_attribute_initialize(
+     libewf_single_file_extended_attribute_t **single_file_extended_attribute,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_free(
+     libewf_single_file_extended_attribute_t **single_file_extended_attribute,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_clone(
+     libewf_single_file_extended_attribute_t **destination_single_file_extended_attribute,
+     libewf_single_file_extended_attribute_t *source_single_file_extended_attribute,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_get_type(
+     libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+     uint8_t *type,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_get_id(
+     libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+     uint32_t *id,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_get_utf8_name_size(
+     libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+     size_t *utf8_string_size,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_get_utf8_name(
+     libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+     uint8_t *utf8_string,
+     size_t utf8_string_size,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_get_utf16_name_size(
+     libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+     size_t *utf16_string_size,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_get_utf16_name(
+     libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+     uint16_t *utf16_string,
+     size_t utf16_string_size,
+     libcerror_error_t **error );
+
+int libewf_single_file_extended_attribute_get_utf8_value_size(
+	 libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+	 size_t *utf8_string_size,
+	 libcerror_error_t **error);
+
+int libewf_single_file_extended_attribute_get_utf8_value(
+	 libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+	 uint8_t *utf8_string,
+	 size_t utf8_string_size,
+	 libcerror_error_t **error);
+
+int libewf_single_file_extended_attribute_get_utf16_value_size(
+	 libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+	 size_t *utf16_string_size,
+	 libcerror_error_t **error);
+
+int libewf_single_file_extended_attribute_get_utf16_value(
+	 libewf_single_file_extended_attribute_t *single_file_extended_attribute,
+	 uint16_t *utf16_string,
+	 size_t utf16_string_size,
+	 libcerror_error_t **error);
+
+#pragma pack(push, 1) 
```

Comment:
why is this defined half way through the source file, move to separate source file.